myia/ir/graph_utils.py

```diff
@@ -0,0 +1,68 @@
+"""Utility functions for myia graph."""
+from myia.ir.node import Apply, Graph
+from myia.utils.directed_graph import DirectedGraph
+
+
+def toposort(graph: Graph, reverse=True):
+    """Return graph nodes in topological order.
+
+    :param graph: myia graph
+    :param reverse: if True, return nodes in execution order (default)
```

Comment:
Hi ! By default, as I visit graph from `return_` node to parameters, I get the topological order from user to used nodes, which is not the execution order. As I guess people would like to get execution order, I set `reverse` to True by default, so that one must explicitly set it to `False` to get raw order (user to used), e.g. if execution order is not necessary, or to save time (skipping reversing).

Should I still set `reverse` to False by default ?

Or, for disambiguation, should I rename `reverse` to something like `execution_order` ?

Reply:
I meant that "reverse" being True should mean the opposite order of whatever the default is, so in this case "reverse" would be from user to used, and you'd have to set reverse to True to get the "raw" order. Basically, if a function has a `reverse` argument I expect it to be `False` by default -- the fact that you have to reverse the raw list to get the execution order is an implementation detail, it doesn't have to correspond to the function's API.

Reply:
OK, updated !